### ISSUE TITLE
Deprecate or remove cstr

### ISSUE THREAD

**Issue by netizen-a** [jasondyoungberg/limine-rs#36]
Recent rust versions allow native c strings to be used so, `cstr!()` is no longer needed

**Comment by lylythechosenone** [comment]
Indeed, our MSRV is 1.77 (the first to support C-string literals), so this should be valid.
